SoftLayer/managers/ordering.py

```diff
@@ -209,15 +241,289 @@ def get_package_by_key(self, package_keyname, mask=None):
+        order.update(extras)
+        order['packageId'] = package['id']
+        order['location'] = location
+        order['quantity'] = quantity
+        order['useHourlyPricing'] = hourly
+
+        if preset_keyname:
+            preset_id = self.get_preset_by_key(package_keyname, preset_keyname)['id']
+            order['presetId'] = preset_id
+
+        price_ids = self.get_price_id_list(package_keyname, item_keynames)
+        order['prices'] = [{'id': price_id} for price_id in price_ids]
+
+        complex_type = complex_type or self._get_complex_type(package_keyname, item_keynames)
+        order['complexType'] = complex_type
+
+        return self.order_svc.verifyOrder(order)
+
+    def place_order(self, package_keyname, location, item_keynames, complex_type=None,
+                    hourly=True, preset_keyname=None, extras=None, quantity=1):
+        """Places an order with the given package and prices.
+
+        This function takes in parameters needed for an order and places the order.
+
+        :param str package_keyname: The keyname for the package being ordered
+        :param str location: The datacenter location string for ordering (Ex: DALLAS13)
+        :param list item_keynames: The list of item keyname strings to order. To see list of
+                                   possible keynames for a package, use list_items()
+                                   (or `slcli order item-list`)
+        :param str complex_type: The complex type to send with the order. Typically begins
+                                 with 'SoftLayer_Container_Product_Order_'. If not specified,
+                                 the complex type will attempt to be derived from
+                                 the package and items, and if unable, a SoftLayerError
+                                 will be raised.
+        :param bool hourly: If true, uses hourly billing, otherwise uses monthly billing
+        :param string preset_keyname: If needed, specifies a preset to use for that package.
+                                      To see a list of possible keynames for a package, use
+                                      list_preset() (or `slcli order preset-list`)
+        :param dict extras: The extra data for the order in dictionary format.
+                            Example: A VSI order requires hostname and domain to be set, so
+                            extras will look like the following:
+                                {'virtualGuests': [{'hostname': 'test',
+                                                    'domain': 'softlayer.com'}]}
+        :param int quantity: The number of resources to order
+
+        """
+        # verify the order, and if the order is valid, the proper prices will be filled
+        # into the order template, so we can just send that to placeOrder to order it
+        verified_order = self.verify_order(package_keyname, location, item_keynames,
```

Comment:
verifyOrder() can strip data out of the original order in the returned order, so calling placeOrder with the verified order will be omitting some of the original order data. verify_order() and place_order() should then have a same base prepare_order() function that it then either calls verify or place on, depending on the function it is in.